src/lib/crypto/kimchi_bindings/stubs/src/pasta_fp_plonk_proof.rs

```diff
@@ -215,28 +215,32 @@ pub fn caml_pasta_fp_plonk_proof_example_with_foreign_field_mul(
     srs: CamlFpSrs,
 ) -> (CamlPastaFpPlonkIndex, CamlProverProof<CamlGVesta, CamlFp>) {
     use ark_ff::Zero;
-    use poly_commitment::srs::{endos, SRS};
     use kimchi::circuits::{
         constraints::ConstraintSystem,
         gate::{CircuitGate, Connect},
-        polynomials::{foreign_field_add::witness::FFOps, foreign_field_mul, range_check},
+        polynomials::foreign_field_mul,
         wires::Wire,
     };
     use num_bigint::BigUint;
     use num_bigint::RandBigInt;
     use o1_utils::{foreign_field::BigUintForeignFieldHelpers, FieldHelpers};
+    use poly_commitment::srs::{endos, SRS};
     use rand::{rngs::StdRng, SeedableRng};
 
     let foreign_field_modulus = Fq::modulus_biguint();
 
     // Layout
-    //      0    ForeignFieldMul   (foreign field multiplication gadget)
-    //      1    Zero              (foreign field multiplication gadget)
-    //      4-7  multi-range-check (left multiplicand)
-    //      8-11 multi-range-check (right multiplicand)
-    //     12-15 multi-range-check (product1_lo, product1_hi_0, carry1_lo)
-    //     16-19 multi-range-check (result range check)
-    //     20-23 multi-range-check (quotient range check)
+    //      0-1  ForeignFieldMul | Zero
+    //      2-5  compact-multi-range-check (result range check)
+    //        6  "single" Generic (result bound)
+    //      7-10 multi-range-check (quotient range check)
+    //     11-14 multi-range-check (quotient_bound, product1_lo, product1_hi_0)
+    //     later limb-check result bound
+    //        15 Generic (left and right bounds)
+    //     16-19 multi-range-check (left multiplicand)
+    //     20-23 multi-range-check (right multiplicand)
+    //     24-27 multi-range-check (result bound, left bound, right bound)
+    // TODO: check when kimchi is merged to berkeley
```

Comment:
Maybe a good idea to explain in this example when the full range-check of `r` would be required.

Reply:
I don't think we want this here. It will only be needed in full ECDSA. To understand when to use a full range check for bounded `r`,  the RFC is the best source of information.